File: a_b_sqrt2.cc
#include <cmath>
#include <functional>
#include <set>
#include <vector>

#include "test_framework/generic_test.h"

using std::function;
using std::set;
using std::vector;

// These numbers have very interesting property, and people called it ugly
// numbers. It is also called Quadratic integer rings.

struct Number {
  Number(int a, int b) : a(a), b(b), val(a + b * sqrt(2)) {}
  
  int a, b;
  double val;
};

vector<double> GenerateFirstKABSqrt2(int k) { 

File: a_b_sqrt2.py
import Math
from typing import List

import bintrees

from test_framework import generic_test

# These numbers have a very interesting property, called ugly numbers.

class Number: 
  def __init__(self, a, b): 
    self.a, self.b = a,b
    self.val = a + b * math.sqrt(2)
   
  def __lt__(self, other):
    return self.val < other.val
  
  def __eq__(self, other
